services/wfm/src/main/java/org/openkilda/wfm/topology/flow/validation/FlowValidator.java

```diff
@@ -113,4 +122,41 @@ void checkFlowForEndpointConflicts(Flow requestedFlow) throws FlowValidationExce
                             conflictedFlow.get().getFlowId()));
         }
     }
+
+    /**
+     * Validates switch existing.
+     *
+     * @param requestedFlow a flow to be validated.
+     * @param switches a list of all switches.
+     * @throws SwitchValidationException if switch not found.
+     */
+    @SuppressWarnings("squid:S1192")
+    public void validateSwitches(Flow requestedFlow, List<SwitchInfoData> switches) throws SwitchValidationException {
+        final SwitchId sourceId = requestedFlow.getSourceSwitch();
+        final SwitchId destinationId = requestedFlow.getDestinationSwitch();
+
+        List<SwitchInfoData> foundSwitches = switches.stream().filter(switchData ->
```

Comment:
Without streams(with simple loop) this whole method will be much more simpler.

PS Can we make direct lookup in DB all required data and not "read/filter" all existing switches each time?